Kafka integration (#291)
overture-stack/maestro: docs/usage.md

Reviewer: the changes added here are good for now, but I think a deeper rewrite is needed as the complexity increases quite quickly, and Kafka behooves some additional context and introductions

i.e. approving with a TODO for @MitchellShiell and myself. Not a blocker for this PR

## Kafka topics

Maestro can be configured to listen to Kafka topics as described in the Running Configurations section.

```yaml
MAESTRO_KAFKA_SERVER=http://kafka:9092
```

> The `MAESTRO_KAFKA_SERVER` configuration specifies the Kafka server(s) to connect to.

Reviewer: this gives the impression you can connect to multiple separate Kafka instances, is that so?
in that case, how many, and how do you specify different urls? (I don't see the usual _1, _2, etc pattern)

Author: updated the documentation with some example how to configure multiple brokers

There are two types of topics that can be configured, depending on your needs:

### Request topics

Request topics are used to send on-demand messages instructing Maestro to fetch the documents from the repository source and perform a specific indexing action.

> Following configuration applies for SONG or Lyric repositories

Reviewer: ```suggestion
The following applies for both SONG and Lyric instances
```
